Functions.cs

```diff
@@ -2,86 +2,112 @@ namespace Company.Function;
+
+        // data for utilities average emissions, values in g per hour
+        readonly Dictionary<string, double> utilities = new()
         {
-            { "Wool", 13.89 },
-            { "Acrylic Fabric", 11.53 },
-            { "Cotton", 8.3 },
-            { "Silk", 7.63 },
-            { "Nylon", 7.31 },
-            { "Polyester", 6.4 },
-            { "Linen", 5.4 }
+            { "Electrivity", 371.03856 }
         };
         // driving calcuation, returns a string of total emissions from varaibles
-        public string DrivingCalculation(string vechileType/*change to type*/, string distance)
+        public double DrivingCalculation(string vechileType, string distance)
         {
-            double vechileTypeEmission = foodEmissions[vechileType];
-            double gallons = int.Parse(distance) / vechileTypeEmission;
-            double litersConversion = literInGallon * gallons;
-            double kgOfCo2 = litersConversion * literToKgCo2;
-            // double roundUp5Decimals = Math.Ceiling(kgOfCo2 * 100000) / 100000;
-            return kgOfCo2.ToString();
+            // find vechile emission
+            double vechileTypeEmission = vehicleType[vechileType];
+            // convert miles to km
+            double distanceKm = double.Parse(distance) * 1.609344;
+            // gram of co2, distance in km * gram per km
+            double gramsOfCo2 = distanceKm * vechileTypeEmission;
+            // convert grams to kg, then kg to lbs
+            double lbsOfCo2 = (gramsOfCo2 / 1000) * 2.20462;
+            // round up 10 decimals
+            return Math.Ceiling(lbsOfCo2 * 10000000000) / 10000000000;
         }
         // food calculations, returns a string of total emissions from variables
-        public string FoodCalculation(string foodName, string amount)
+        public double FoodCalculation(string foodName, string amount)
         {
-            double emissionPerUnitOfFood = foodEmissions[foodName];
-            double kgOfCo2 = emissionPerUnitOfFood * int.Parse(amount);
-            // double roundUp5Decimals = Math.Ceiling(kgOfCo2 * 100000) / 100000;
-            return kgOfCo2.ToString();
+            // get food info for food name
+            FoodEmission foodInfo = foodEmissions[foodName];
+            // get the grams and isPerServing from FoodEmission
+            int grams = foodInfo.Emission;
+            bool isPerServing = foodInfo.IsPerServing;
+
+            double gramsOfCo2;
+            if (isPerServing) {
+                // the amount, refers to per serving
+                gramsOfCo2 = grams * double.Parse(amount);
+            } else {
+                // amount, refers to grams
+                double amountPerHundredGrams = double.Parse(amount) / 100.0; // 100 because each grams info is per 100 g
+                gramsOfCo2 = grams * amountPerHundredGrams;
+            }
+            // convert grams to kg, then kg to lbs
+            double lbsOfCo2 = (gramsOfCo2 / 1000) * 2.20462;
```

Comment:
The last two line of each calculation function seems repeating. Can you put them into one function like ConvertGramToPounds? Have the amount of digit as a parameter as well.